### PR TITLE
Replace Path with URI

### LINKED ISSUE #398: Bad path error connecting to Dropbox repository from Windows
Created a Dropbox repository in OS X.

Tried connecting to it in Windows.  Was able to connect to repository, but get error when downloading files.

Appears to be an issue with the path name being passed to Dropbox.
```
syncany down -d

debug
Application version: 0.4.3-alpha
3-3-15 15:39:44.841 | CommandLineClie | main           | INFO : Application version: 0.4.3-alpha
3-3-15 15:39:44.904 | PidFileUtil     | main           | INFO : isProcessRunningWindows(3932) returned java.exe
             3932 Console                    1    105,996 K, process running = true
3-3-15 15:39:44.904 | org.syncany.plu | main           | FINE : Validating required fields
3-3-15 15:39:44.919 | org.syncany.plu | main           | FINE : Validating required fields
3-3-15 15:39:44.919 | DatabaseConnect | main           | INFO : Loading database driver org.hsqldb.jdbcDriver ...
3-3-15 15:39:45.232 | hsqldb.db.HSQLD | main           | INFO : dataFileCache open start
3-3-15 15:39:45.232 | hsqldb.db.HSQLD | main           | INFO : dataFileCache open end
3-3-15 15:39:45.299 | DownOperation   | main           | INFO :
3-3-15 15:39:45.299 | DownOperation   | main           | INFO : Running 'Sync down' at client YOjLwNSPhpTJvzajfgfB ...
3-3-15 15:39:45.299 | DownOperation   | main           | INFO : --------------------------------------------
3-3-15 15:39:45.299 | InternalEventBu | main           | INFO : Event bus 'LocalEventBus': Posting event DownStartSyncEx
ternalEvent (org.syncany.operations.daemon.messages.DownStartSyncExternalEvent@7048535f) ...
3-3-15 15:39:45.315 | LsRemoteOperati | main           | INFO :
3-3-15 15:39:45.315 | LsRemoteOperati | main           | INFO : Running 'Remote Status' at client YOjLwNSPhpTJvzajfgfB .
..
3-3-15 15:39:45.315 | LsRemoteOperati | main           | INFO : --------------------------------------------
3-3-15 15:39:45.315 | InternalEventBu | main           | INFO : Event bus 'LocalEventBus': Posting event LsRemoteStartSy
ncExternalEvent (org.syncany.operations.daemon.messages.LsRemoteStartSyncExternalEvent@604b1e1d) ...
Checking remote changes ...3-3-15 15:39:45.331 | LsRemoteOperati | main           | INFO : Retrieving remote database li
st.
3-3-15 15:39:45.331 | CommandLineClie | main           | SEVE : Command org.syncany.cli.DownCommand@7225790e FAILED.
java.lang.IllegalArgumentException: 'path': bad path: must start with "/": "\\synctest\\databases"
        at com.dropbox.core.DbxPath.checkArg(DbxPath.java:48)
        at com.dropbox.core.DbxClient.getMetadataWithChildrenBase(DbxClient.java:174)
        at com.dropbox.core.DbxClient.getMetadataWithChildren(DbxClient.java:148)
        at org.syncany.plugins.dropbox.DropboxTransferManager.list(DropboxTransferManager.java:247)
        at org.syncany.plugins.transfer.RetriableTransferManager$8.execute(RetriableTransferManager.java:138)
        at org.syncany.plugins.transfer.RetriableTransferManager.retryMethod(RetriableTransferManager.java:197)
        at org.syncany.plugins.transfer.RetriableTransferManager.list(RetriableTransferManager.java:135)
        at org.syncany.plugins.transfer.TransactionAwareTransferManager.list(TransactionAwareTransferManager.java:139)
        at org.syncany.operations.ls_remote.LsRemoteOperation.listUnknownRemoteDatabases(LsRemoteOperation.java:95)
        at org.syncany.operations.ls_remote.LsRemoteOperation.execute(LsRemoteOperation.java:78)
        at org.syncany.operations.down.DownOperation.listUnknownRemoteDatabases(DownOperation.java:265)
        at org.syncany.operations.down.DownOperation.checkPreconditions(DownOperation.java:240)
        at org.syncany.operations.down.DownOperation.execute(DownOperation.java:137)
        at org.syncany.Client.down(Client.java:125)
        at org.syncany.cli.DownCommand.execute(DownCommand.java:55)
        at org.syncany.cli.CommandLineClient.runLocally(CommandLineClient.java:389)
        at org.syncany.cli.CommandLineClient.runCommand(CommandLineClient.java:379)
        at org.syncany.cli.CommandLineClient.start(CommandLineClient.java:195)
        at org.syncany.Syncany.main(Syncany.java:55)
Error: 'path': bad path: must start with "/": "\\synctest\\databases"
       Refer to help page using '--help'.
```

### PR DESCRIPTION
**Pull request by cr0**
fixes syncany/syncany#398

### CODE REVIEW

File: src/main/java/org/syncany/plugins/dropbox/DropboxTransferManager.java [hunk 1]
```diff
@@ -389,4 +389,12 @@ public boolean testRepoFileExists() {
 			return false;
 		}
 	}
+
+	private URI makeChildURI(URI parent, String child) {
```
Comment (binwiederhier) [inline src/main/java/org/syncany/plugins/dropbox/DropboxTransferManager.java]: How about `createUri()`?

File: src/main/java/org/syncany/plugins/dropbox/DropboxTransferManager.java [hunk 2]
```diff
@@ -389,4 +389,12 @@ public boolean testRepoFileExists() {
...
+		if (!parent.toString().endsWith("/")) {
+			parent = URI.create(parent.toString() + "/");
+		}
+		return parent.resolve(child);
```
Comment (binwiederhier) [inline src/main/java/org/syncany/plugins/dropbox/DropboxTransferManager.java:398]: You know I'm a fan of spaces and newlines. Would you mind adding one before the `return`?

**Comment by cr0** [comment]
@binwiederhier the build is failing because of SonarRunner. Could you have a look once you've some time.

**Comment by binwiederhier** [comment]
Merge to develop at will after my small comments are applied. 
Once this is done I can do a release.

**Comment by binwiederhier** [comment]
Urgh! Sure thing ...

**Comment by binwiederhier** [comment]
@MrHug @cr0 This seems to describe a workaround. It'd be great if you guys could take care of that: https://issues.gradle.org/browse/GRADLE-1813

Please make sure to leave a comment next to the solution with the above link, so that we don't run into this issue again. Thanks!!

**Comment by MrHug** [comment]
I'll take a look at this tonight/tomorrow :wink:

**Comment by cr0** [comment]
Do we need to create a separate issue for that? Do we furthermore need to modify the main project or is this plugin related?

**Comment by MrHug** [comment]
I think we need to change the main project, as I believe it goes wrong when
using these gradle configs for these as well, but after finishing up a
document on Synany's features for the project, I'll take a look at this. A
seperate issue might be best/easiest :wink:

Op zo 8 mrt. 2015 om 14:59 schreef Christian Roth <notifications@github.com

**Pull request merged**